### PR TITLE
Update model$Margins and modify PUR conversion method

### PR DESCRIPTION
**Pull request by MoLi7**

### CODE REVIEW

File: R/Margins.R [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
-    MarginsTable <- rbind(PCE, PEQ)
+    MarginsTable <- useeior::Detail_Margins_2012_BeforeRedef
   }
+  # Remove Imports records
+  MarginsTable <- MarginsTable[MarginsTable$NIPACode!="F05000", ]
+  # Remove Scrap, Used and secondhand goods, and Rest of world adjustment commodities
+  MarginsTable <- MarginsTable[!MarginsTable$CommodityCode%in%c("S00401", "S00402", "S00900"), ]
+  # Convert negative values to non-negative
+  value_columns <- c("ProducersValue", "Transportation", "Wholesale", "Retail")
+  MarginsTable[, value_columns] <- abs(MarginsTable[, value_columns])
```
Comment (MoLi7) [inline R/Margins.R:15]: @WesIngwersen here are the record removal and flipping sign of negative PRO.

File: R/Margins.R [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
   MarginsTable <- MarginsTable[match(model$Commodities$Code, MarginsTable$CommodityCode), ]
   # Transform MarginsTable from Commodity to Industry format
   if (model$specs$CommoditybyIndustryType=="Industry") {
     # Generate a commodity x industry commodity mix matrix, see Miller and Blair section 5.3.2
     CommodityMix <- generateCommodityMixMatrix(model)
-    MarginsTable_Industry <- as.data.frame(model$Industries)
+    MarginsTable_Industry <- model$Industries[, "Code", drop = FALSE]
     colnames(MarginsTable_Industry) <- "IndustryCode"
     # Transform ProducerValue from Commodity to Industry format
     # ! Not transforming Transportation, Wholesale and Retail to Industry format now
```
Comment (MoLi7) [inline R/Margins.R:36]: @WesIngwersen TWR are not transformed to Industry format here or anywhere else. We should consider doing it in a proper way.

**Comment by MoLi7** [comment]
Since we decided to not use PCE and PEQ bridge tables for margin calculations, shall we remove them from useeior? @WesIngwersen

File: R/Margins.R (hunk 2 above)
Comment (WesIngwersen) [reply R/Margins.R:36]: I don't think that the margin components themselves need be transformed into industry format

**Comment by WesIngwersen** [comment]
> Since we decided to not use PCE and PEQ bridge tables for margin calculations, shall we remove them from useeior? @WesIngwersen

Yes, following the principle of removing any data not used in the package

File: R/Margins.R (hunk 1 above)
Comment (WesIngwersen) [reply R/Margins.R:15]: I think we only want to use abs() for the `ProducersValue`.  The margins columns as I have see are all positive -- if they are not I am not sure we'd want to transform them

**Review by WesIngwersen — COMMENTED** [review]
Let's leave this open until we get more feedback and test it in the meantime.

File: R/Margins.R (hunk 1 above)
Comment (WesIngwersen) [reply R/Margins.R:15]: Well I see that margins are negative when change in inventories in positive..which means adding to stock. I think its safest to remove this as well.

**Comment by WesIngwersen** [comment]
In general I am opposed to using sector codes in R code because this forces the R code to be schema specific. These sectors with special treatment should be identified via the schema, and not by name

File: R/Margins.R (hunk 1 above)
Comment (WesIngwersen) [reply R/Margins.R:15]: I will do that (remove change in inventories)

Comment (WesIngwersen) [reply R/Margins.R]: Done in fix_margin@5237c935

Comment (WesIngwersen) [reply R/Margins.R]: > I think we only want to use abs() for the `ProducersValue`. The margins columns as I have see are all positive -- if they are not I am not sure we'd want to transform them

I did this as well in usepa/useeior@8a560cdf

**Comment by MoLi7** [comment]
> In general I am opposed to using sector codes in R code because this forces the R code to be schema specific. These sectors with special treatment should be identified via the schema, and not by name

I agree. We should use Schema_Info.csv ([example](https://github.com/USEPA/useeior/blob/master/inst/extdata/2012_Detail_Schema_Info.csv)) to specify sector codes. We need to add Used and secondhand goods, Non-comparable imports, and Rest of world adjustment as columns in each Schema_Info.csv.

**Comment by WesIngwersen** [comment]
> > In general I am opposed to using sector codes in R code because this forces the R code to be schema specific. These sectors with special treatment should be identified via the schema, and not by name
> 
> I agree. We should use Schema_Info.csv ([example](https://github.com/USEPA/useeior/blob/master/inst/extdata/2012_Detail_Schema_Info.csv)) to specify sector codes. We need to add Used and secondhand goods, Non-comparable imports, and Rest of world adjustment as columns in each Schema_Info.csv.

Thanks but i think perhaps we can use groupings like at the Summary level - `Used/scrap` and `RoW/Noncomparable' if we treat the sectors in those groupings the same in all cases.   I think its really the treatment that would define the level of resolution of those groupings in the schema

**Comment by MoLi7** [comment]
> Thanks but i think perhaps we can use groupings like at the Summary level - `Used/scrap` and `RoW/Noncomparable' if we treat the sectors in those groupings the same in all cases. I think its really the treatment that would define the level of resolution of those groupings in the schema

What about the case in which we need to specify whether 'Scrap' is included? See [here](https://github.com/USEPA/useeior/blob/894eba1847dbd034c50680ea239ca4d1d2e86ae3/inst/extdata/USEEIOv2.0.yml#L12) in USEEIOv2.0.yml.
The grouping will not be able to separate 'Scrap' from 'Used and secondhand goods' (at Detail level), unless we want to treat Scrap and Used in the same way, i.e. to include or exclude the two together.

**Comment by WesIngwersen** [comment]
> > Thanks but i think perhaps we can use groupings like at the Summary level - `Used/scrap` and `RoW/Noncomparable' if we treat the sectors in those groupings the same in all cases. I think its really the treatment that would define the level of resolution of those groupings in the schema
> 
> What about the case in which we need to specify whether 'Scrap' is included? See [here](https://github.com/USEPA/useeior/blob/894eba1847dbd034c50680ea239ca4d1d2e86ae3/inst/extdata/USEEIOv2.0.yml#L12) in USEEIOv2.0.yml.
> The grouping will not be able to separate 'Scrap' from 'Used and secondhand goods' (at Detail level), unless we want to treat Scrap and Used in the same way, i.e. to include or exclude the two together.

We actually have not implemented the old method of reallocating scrap (and I don't think we should at this point). But they do have slightly different treatment in the Make Table, so I do think it is OK to them separate (even though they are combined at other schema levels and this might present an issue in model building).

**Comment by MoLi7** [comment]
> We actually have not implemented the old method of reallocating scrap (and I don't think we should at this point). But they do have slightly different treatment in the Make Table, so I do think it is OK to them separate (even though they are combined at other schema levels and this might present an issue in model building).

So the conclusion is "keeping them separate in Schema_Info of [Detail](https://github.com/USEPA/useeior/blob/fix_margin/inst/extdata/2012_Detail_Schema_Info.csv), [Summary](https://github.com/USEPA/useeior/blob/fix_margin/inst/extdata/2012_Summary_Schema_Info.csv) and [Sector](https://github.com/USEPA/useeior/blob/fix_margin/inst/extdata/2012_Sector_Schema_Info.csv)"? Or just separate in Detail_Schema_Info? Though I think we should keep them separate in all Schema_Info tables so that the same fields can assure standardized code in model building.

**Comment by MoLi7** [comment]
@WesIngwersen I think the fix of margins is completed and we are ready to merge the PR.

**Pull request merged**